.github/workflows/e2e_test.yml

```diff
@@ -0,0 +1,39 @@
+name: "End-to-End Testing"
+
+on:
+  pull_request: # empty object will use defaults
+  push:
+    branches:
+      - "main"
+
+jobs:
+  end-to-end-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Install Playwright browsers
+        run: npx playwright install --with-deps
+
+      - name: Build site and run end-to-end tests
+        run: npm run test:e2e
+
+      - name: "Upload test report"
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: e2e-test-report
+          path: e2e-tests/report/
+          retention-days: 30
```

Comment:
Didn't know about this option until I was working on this; figure we'll probably do the same for unit test coverage reports. Let me know if y'all have any feelings about the `retention-days` limit I've set; a month feels sufficient but I'm down to increase it.

It's a bit of a pain because you have to download the file and then run `npx playwright show-report` in order to view it (instead of GitHub hosting it and just clicking a link), but since this shouldn't be a problem too often I figure this is fine for now.

Reply:
This looks good for now! I think it would be nice as a followup to upload it to the final site preview under something like `/.meta/playwright-report` so we can link to the report directly within the PR and keep the report around for as long as the deployment exists. We should be able to use the artifact upload/download to do so, so this PR sets us up to do that in the future.

Reply:
Yeah I'll be handling adding code coverage reports to the build in #67 / #68, so I might handle this there as well.

I'm curious how you think we should go about doing this; brute-force would be to run the unit tests and end-to-end tests as part of the Vercel build command (that way all the necessary files are immediately available to Vercel), but there's got to be a better way right? Ideally we'd be able to fetch artifacts from GitHub as part of the Vercel build process but I suppose we'd have to delay the Vercel build, maybe only triggering it once tests are completed. Maybe there's a simpler solution here that I'm just not seeing?